keanu-project/src/test/java/io/improbable/keanu/algorithms/variational/optimizer/gradient/testcase/RosenbrockTestCase.java

```diff
@@ -0,0 +1,55 @@
+package io.improbable.keanu.algorithms.variational.optimizer.gradient.testcase;
+
+import io.improbable.keanu.algorithms.variational.optimizer.OptimizedResult;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.UniformVertex;
+
+import static junit.framework.TestCase.assertEquals;
+
+/**
+ * This is a common test case for gradient optimization algorithms.
+ * <p>
+ * https://en.wikipedia.org/wiki/Rosenbrock_function
+ * <p>
+ * f(x,y) = (a - x)^2 + b*(y - x^2)^2
+ * <p>
+ * The function has been inverted here because our optimizers find maxima and this
+ * function provides a global minima. The global maxima is at (a, a^2) so for a=1,b=100
+ * the maxima will be (1,1)
+ */
+public class RosenbrockTestCase extends BivariateFunctionTestCase {
+
+    private final double expectedX, expectedY;
+
+    public RosenbrockTestCase(double aParameter, double bParameter, double expectedX, double expectedY) {
```

Comment:
The expected values for this test can be calculated from `aParameter`. Why don't we just do that calculation instead of passing it in? Would make it a bit more clearer how the expected is calculated rather than being magic numbers.

Reply:
Very good point. Done.